django/db/migrations/state.py

```diff
@@ -353,6 +353,12 @@ def __init__(self, app_label, name, fields, options=None, bases=None, managers=N
                     'ModelState.fields cannot refer to a model class - "%s.through" does. '
                     'Use a string reference instead.' % name
                 )
+        # Sanity-check that indexes have their names set.
+        for index in self.options['indexes']:
+            try:
+                getattr(index, 'name')
+            except AttributeError:
+                raise ValueError('Indexes passed to ModelState should have their name set.')
```

Comment:
I agree that this check is required in `AddIndex` (and hence I have done that in my recent commit related to migrations) but I would like to know your thoughts on keeping a sanity-check here, because in the `autodetector` there is no use of a `ModelState` object with an index which doesn't have a name AKAIK (`name` is used in `Index.__eq__`). There might be other use cases of `ModelState` objects other than in `autodetector` that I am unaware of, in which case we might not want to keep this check.